datafusion/optimizer/src/optimizer.rs

```diff
@@ -237,6 +238,7 @@ impl Optimizer {
         let rules: Vec<Arc<dyn OptimizerRule + Sync + Send>> = vec![
             Arc::new(InlineTableScan::new()),
             Arc::new(TypeCoercion::new()),
+            Arc::new(ExtractEquijoinPredicate::new()),
```

Comment:
Some subquery will be converted to the join in the following up `SubqueryFilterToJoin` rule., and I am not sure if this rule is right in this position

Reply:
cc @ygf11 @jackwener

Reply:
Yes, this rule should be placed behind `SubqueryFilterToJoin` rule. @jackwener also suggested about it in https://github.com/apache/arrow-datafusion/pull/4725#discussion_r1056956986. 

But I think it is ok now, because `SubqueryFilterToJoin` and other similar rule directly specify the equijoin predicate, so they do not depends `ExtractEquijoinPredicate` currently. 

I plan to do this movement after #4725 merged.

Reply:
> Yes, this rule should be placed behind `SubqueryFilterToJoin` rule. @jackwener also suggested about it in [#4725 (comment)](https://github.com/apache/arrow-datafusion/pull/4725#discussion_r1056956986).
> 
> I think it is ok now, because `SubqueryFilterToJoin` and other similar rule directly specify the equijoin predicate, so they do not depends `ExtractEquijoinPredicate` currently.

Yes, other rule will convert query to join with specified equal-join predicate and non-equal-join predicate.

> I plan to do this movement after #4725 merged.

LGTM